### PR TITLE
Search Users by Last Name

### PR DESCRIPTION
**Pull request by RamuKammagani-osi**

### CODE REVIEW

File: app/javascript/home/home.js [hunk 1]
```diff
@@ -8,8 +8,16 @@ import { connect } from 'react-redux';
 import { selectUserRecords } from '../selectors/userListSelector';
 
 class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchKey: '',
+    };
+    this.handleTextChange = this.handleTextChange.bind(this);
```
Comment (Sushmareddy-OSI) [inline app/javascript/home/home.js]: No Need to bind.

File: app/controllers/api/user_list_controller.rb [hunk 2]
```diff
@@ -2,8 +2,13 @@
 
 module Api
   class UserListController < ActionController::API
+    # def index
+    #   users = Users::UserRepository.new.get_users(session[:token])
+    #   render json: users
+    # end
+
```
Comment (Sushmareddy-OSI) [inline app/controllers/api/user_list_controller.rb]: can be removed

**Comment by RamuKammagani-osi** [comment]
retest this please

**Comment by RamuKammagani-osi** [comment]
retest this please

**Comment by chadbratton** [comment]
retest this please

File: lib/infrastructure/http_service.rb [hunk 3]
```diff
@@ -8,8 +8,8 @@ def initialize(base_url = Rails.configuration.micro_services['perry_api_base_url
       @base_url = base_url
     end
 
-    def get(url, token)
-      http_connection.get("#{url}?token=#{token}")
+    def get(url, last_name, token)
```
Comment (kelfink) [inline lib/infrastructure/http_service.rb]: We do NOT want to add these parameters into the http_service.  I believe this is being addressed by the next story.  Either this needs to be fixed now, or I need you to add a refactor story for the next sprint that undoes this.  The parameters here need to be generic, and the url with query string must be constructed by the caller.

File: lib/users/user_repository.rb [hunk 4]
```diff
@@ -6,14 +6,14 @@ def initialize(http_service = Infrastructure::HttpService.new)
       @http_service = http_service
     end
 
-    def get_users(token)
-      response = @http_service.get('/perry/idm/users', token)
+    def get_users(last_name, token)
+      response = @http_service.get('/perry/idm/users', last_name, token)
```
Comment (kelfink) [inline lib/users/user_repository.rb]: This is where the url needs to be constructed, by adding the last name onto the url string.  We might add logic to the http service which correctly adds the token whether the url has a query string already, or not.

File: lib/infrastructure/http_service.rb (hunk 3 above)
Comment (RamuKammagani-osi) [reply lib/infrastructure/http_service.rb]: Right! I have already a story in next sprint, Will fix that as soon as possible, Thanks

File: lib/users/user_repository.rb (hunk 4 above)
Comment (RamuKammagani-osi) [reply lib/users/user_repository.rb]: 👍

**Comment by kelfink** [comment]
With a couple of refactors to be done in the next sprint, this is OK to go.  We will undo changes to http_service and perform url manipulations in the calling code instead.

**Pull request merged**